system-dependencies.sh

```diff
@@ -320,21 +320,44 @@ function download_xcode_platforms ()
 		return
 	fi
 
-	TVOS_SIMULATOR_VERSION=$(/usr/libexec/PlistBuddy -c 'Print :ProductBuildVersion' "$XCODE_DEVELOPER_ROOT"/Platforms/AppleTVSimulator.platform/version.plist)
-	WATCHOS_SIMULATOR_VERSION=$(/usr/libexec/PlistBuddy -c 'Print :ProductBuildVersion' "$XCODE_DEVELOPER_ROOT"/Platforms/WatchSimulator.platform/version.plist)
+	local SIMULATOR_RUNTIMES
+	SIMULATOR_RUNTIMES=$("$XCODE_DEVELOPER_ROOT"/usr/bin/simctl runtime list)
 
-	if test -d "/Library/Developer/CoreSimulator/Volumes/tvOS_$TVOS_SIMULATOR_VERSION/Library/Developer/CoreSimulator/Profiles/Runtimes/tvOS $TVOS_VERSION.simruntime"; then
-		if test -d "/Library/Developer/CoreSimulator/Volumes/watchOS_$WATCHOS_SIMULATOR_VERSION/Library/Developer/CoreSimulator/Profiles/Runtimes/watchOS $WATCHOS_VERSION.simruntime"; then
-			log "All the additional platforms have already been downloaded for this version of Xcode ($XCODE_VERSION)"
-			return
-		fi
+	TVOS_SIMULATOR_RUNTIME=$(echo "$SIMULATOR_RUNTIMES" | grep "^tvOS $TVOS_VERSION .*Ready" || true)
+	WATCHOS_SIMULATOR_RUNTIME=$(echo "$SIMULATOR_RUNTIMES" | grep "^watchOS $WATCHOS_VERSION .*Ready" || true)
+
+	MUST_INSTALL_RUNTIMES=
+	if test -z "$TVOS_SIMULATOR_RUNTIME"; then
+		MUST_INSTALL_RUNTIMES=1
+	fi
+	if test -z "$WATCHOS_SIMULATOR_RUNTIME"; then
+		MUST_INSTALL_RUNTIMES=1
+	fi
+	if test -z "$MUST_INSTALL_RUNTIMES"; then
+		log "All the additional platforms have already been downloaded for this version of Xcode ($XCODE_VERSION)"
+		log "    $TVOS_SIMULATOR_RUNTIME"
+		log "    $WATCHOS_SIMULATOR_RUNTIME"
+		return
 	fi
 
-	if ! test -z "$PROVISION_XCODE"; then
-		fail "Xcode has additional platforms that must be downloaded. Execute './system-dependencies.sh --provision-xcode' to execute those tasks."
+	if test -z "$PROVISION_XCODE"; then
```

Comment:
Looks like the conditional was switched, was this intentional?

Reply:
Yes, that's this part of the PR description:

> Also fix a condition that would cause the script to provision simulator
runtimes as long as --provision-xcode was *not* passed to the script.